### PR TITLE
[lte][agw] Support to test pco option ipcp

### PR DESCRIPTION
**Pull request by pruthvihebbani**
[lte][agw] Support to test pco option ipcp

## Summary

- Added support in s1ap utility to include pco option ipcp
- Added a new s1 sim testcase to test ipcp
- Minor change in the debug log

## Test Plan

Tested that s1sim includes ipcp option in PCO IE in pdn connectivity request message during attach procedure and MME sends primary and secondary dns ip addresses in the PCO IE in activate default eps bearer context message

### CODE REVIEW

**Comment by rdefosse** [comment]
MAGMA-OAI-TESTS

**Comment by rdefosse** [comment]
MAGMA-OAI-TESTS

**Comment by ulaskozat** [comment]
@pruthvihebbani is this still work in progress or something ready to be reviewed?
@ulaskozat its WIP because after adding support in s1ap utils to send ipcp option in pco, I could not reproduce the issue you were facing. The TC I have added works fine. So I wanted to crosscheck the pcap and logs of the issue you are facing to see if they match with my changes. I had asked you to share the issue logs.

File: lte/gateway/c/oai/tasks/sgw/spgw_state.cpp [hunk 1]
```diff
@@ -82,7 +82,8 @@ void sgw_free_s11_bearer_context_information(
   if (*context_p) {
     sgw_free_pdn_connection(
         &(*context_p)->sgw_eps_bearer_context_information.pdn_connection);
-
+    clear_protocol_configuration_options(
+        &(*context_p)->sgw_eps_bearer_context_information.saved_message.pco);
```
Comment (ulaskozat) [inline lte/gateway/c/oai/tasks/sgw/spgw_state.cpp:93]: how come we have not seen this memory leak before?

Comment (pruthvihebbani) [reply lte/gateway/c/oai/tasks/sgw/spgw_state.cpp:93]: What we had till now in PCO options were all container IDs. For container ID, the request does not include any data. Data is included only in the response. But ipcp is a protocol id and in this case request also contains data and needs to be freed

**Review by ulaskozat — APPROVED** [review]
LGTM

**Comment by ulaskozat** [comment]
@pruthvihebbani I removed {WIP} from the title and let's land this.

File: lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py [hunk 2]
```diff
@@ -280,9 +313,9 @@ def attach(
         attach_req.pdnType_pr.pdn_type = pdn_type
 
         # Populate PCO only if pcscf_addr_type is set
```
Comment (ulaskozat) [inline lte/gateway/python/integ_tests/s1aptests/s1ap_utils.py]: modify the comment line

**Review by ardzoht — APPROVED** [review]

**Comment by ekowtaylor** [comment]
@ulaskozat looks like this is ready to me merged and closed, can we move it along?

**Pull request merged**